lib/plugins/crawler/index.js

```diff
@@ -68,7 +89,7 @@ module.exports = {
 
             if (pageCount >= maxPages) {
               log.info('Crawler stopped after %d urls', pageCount);
-              crawler.stop();
+              crawler.stop(true);
```

Comment:
Even before my changes, when specifying e.g. `--crawler.maxPages 5` most of the times I would get 5 pages in the report, but sometime I'd get 6. I guess that's because the crawler already discovered the 6th URL by the time `crawler.stop()` was called. I hope adding `true` will fix that. See https://github.com/simplecrawler/simplecrawler/blob/be994f01bc2b055aabb192290babe647f7a08ca1/lib/crawler.js#L1829

If that's not good enough, an additional pageCount check needs to be added at the top of the  `crawler.on('fetchcomplete', ...)` callback, I guess.